web/src/narrow.js

```diff
@@ -172,10 +172,19 @@ export function update_narrow_title(filter) {
     notifications.redraw_title();
 }
 
+export function handle_empty_narrow_banner() {
+    // Checks if there are no messages and shows the empty narrow message
+    if (message_lists.current.empty()) {
+        narrow_banner.show_empty_narrow_message();
+    } else {
+        narrow_banner.hide_empty_narrow_message();
+    }
+}
+
 export function reset_ui_state() {
     // Resets the state of various visual UI elements that are
     // a function of the current narrow.
-    narrow_banner.hide_empty_narrow_message();
+    handle_empty_narrow_banner();
```

Comment:
This changes the behavior for narrows (a simple stream narrow is an example), which I don't think we want. The empty narrow banner appears before the web app has received the data from the server for the narrow.

You can see the different behavior between this and main if you use the dev tools in Chrome to throttle the network connection to 3G or Good 3G. In main, the empty narrow doesn't show up until the server response; you only see the logo with the spinner. With these changes, you see the empty narrow banner immediately while the logo with the spinner is waiting for the server response.

Reply:
After following the steps you provided, I was able to obtain identical outcomes. 

Additionally, I observed that when we switch from a stream with messages to an empty one, the loading-spinner stops before the narrow banner is displayed. However, if we switch from one empty stream to another, the banner displays prior to the spinner coming to a halt.

https://user-images.githubusercontent.com/96344003/225755380-6f9652ac-acde-4539-9ce7-80b01b8a23ee.mov

Reply:
Yeah, the current behavior where the banner is removed and then reappears if/when no messages are received from the server should be maintained.